File: wild_pointer_fixed.c
#include <stdlib.h>
#include <stdio.h>

int main()
{
    int *ptr = malloc(sizeof *ptr);
    *ptr = 123;
    printf("ptr = %p (%d)\n", ptr, *ptr);

    char *ptr2 = malloc(sizeof *ptr2);
    *ptr2 = 'A';
    printf("ptr = %p (%c)\n", ptr2, *ptr2);
}

File: wild_pointer_fixed.rs



fn main()
{
    let ptr: Box<i32>;
    ptr = Box::new(123);
    println!("ptr = {:p} ({})", ptr, *ptr);

    let ptr2: Box<char>;
    ptr2 = Box::new('A');
    println!("ptr2 = {:p} ({})", ptr2, *ptr2);
}
